### DISCUSSION TITLE
Any way to make <w3m-network-button /> switch chains and set chain.id even when user's wallet is disconnected?

### DISCUSSION THREAD

**Discussion by dappbuildernet** [WalletConnect/walletconnect-monorepo#3780] [Developer Support]
Hi,

I am using web3modal with it's default react+wagmi config (defaultConfig) and hooks work amazingly even when user has not connected yet - blockchain data is loading though I have to manually set chain.id in the hooks arguments (chainid = ....)

Is there any way to switch between preloaded chains without connecting a wallet and get chain.id?

**Comment by glitch-txs** [comment]
Hi, what do you mean by this:
> I have to manually set chain.id in the hooks arguments
could you share a code snippet?

have you tried using https://wagmi.sh/react/hooks/useSwitchNetwork ?

**Reply by dappbuildernet** [reply]
Hi,

I mean that there is no "chain" instance until a wallet is connected. 

But hooks still can get blockchain data using built-in RPCs, even without "chain" existing but with manually set "chainid: ..." parameter.

Is it possible anyhow to switch chain.id with native <w3m-network...> button or another way **without user's wallet being connected**? 

Actually there is a chain (default RPC/WS) even without connecting, so there must be a way to retrieve it's chain property and chain id?

_have you tried using https://wagmi.sh/react/hooks/useSwitchNetwork ?_

It won't work without a user's wallet connected, right? Because it will ask user to switch network in the wallet?

Thank you!

**Reply by glitch-txs** [reply]
which hook are you refering to? usually you can pass the chain to those
